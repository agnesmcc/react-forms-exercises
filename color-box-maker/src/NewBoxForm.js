import React from "react";

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        width: "",
        height: "",
        color: ""
    }
    const [formData, setFormData] = React.useState(INITIAL_STATE)

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formData)
        addBox({...formData})
        setFormData(INITIAL_STATE)
        document.getElementById('width').focus();
    }   

    const handleChange = (e) => {
        const {name, value} = e.target
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }   

    return (
        <div>   
            <form onSubmit={handleSubmit}>
                <div>
                    <label htmlFor="width">Width</label>
                    <input id="width" name="width" type="text" 
                        value={formData.width} 
                        onChange={handleChange}/>
                </div>  
                <div>   
                    <label htmlFor="height">Height</label>
                    <input id="height" name="height" type="text" 
                        value={formData.height}
                        onChange={handleChange}/>
                </div>
                <div>
                    <label htmlFor="color">Color</label>
                    <input id="color" name="color" type="text" 
                        value={formData.color}
                        onChange={handleChange}/>
                </div>
                <button>Add a new box</button>
            </form>
        </div>
    )
}   

export default NewBoxForm;
