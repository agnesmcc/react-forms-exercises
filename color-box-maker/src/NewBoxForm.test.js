import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it('renders without crashing', () => {
  const { getByText } = render(<NewBoxForm />);
  const input = getByText(/Width/);
  expect(input).toBeInTheDocument();
})

it('matches the snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
}) 
