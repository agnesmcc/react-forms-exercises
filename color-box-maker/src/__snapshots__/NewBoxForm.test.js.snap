// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          type="text"
          value=""
        />
      </div>
      <button>
        Add a new box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
